src/runtime/virtcontainers/kata_agent.go

```diff
@@ -1662,6 +1668,60 @@ func (k *kataAgent) createBlkStorageObject(c *Container, m Mount) (*grpc.Storage
 	return vol, err
 }
 
+// handleSMBMounts will create a unique destination mountpoint in the guest for each volume in the
+// given container and will update the OCI spec to utilize this mount point as the new source for the
+// container volume. The container mount structure is updated to store the guest destination mountpoint.
+func (k *kataAgent) handleSMBMounts(c *Container, spec *specs.Spec) ([]*grpc.Storage, error) {
+	var volumeStorages []*grpc.Storage
+	for i, m := range c.mounts {
+		// Handle only cifs mounts
+		if m.Type != "cifs" {
+			continue
+		}
+		// Create Storage
+		vol, err := k.createSMBVolumeObject(c, m)
+		if vol == nil || err != nil {
+			return nil, err
+		}
+
+		// Each device will be mounted at a unique location within the VM only once. Mounting
+		// to the container specific location is handled within the OCI spec. Let's ensure that
+		// the storage mount point is unique for each device. This is then utilized as the source
+		filename := b64.URLEncoding.EncodeToString([]byte(vol.Source))
+		path := filepath.Join(kataGuestSandboxStorageDir(), filename)
+
+		// Update applicable OCI mount source
+		for idx, ociMount := range spec.Mounts {
+			if ociMount.Destination != vol.MountPoint {
+				continue
+			}
+			k.Logger().WithFields(logrus.Fields{
+				"original-source": ociMount.Source,
+				"new-source":      path,
+			}).Debug("Replacing OCI mount source")
+			spec.Mounts[idx].Source = path
+			break
+		}
+
+		// Update storage mountpoint, and save guest device mount path to container mount struct:
+		vol.MountPoint = path
+		c.mounts[i].GuestDeviceMount = path
+		volumeStorages = append(volumeStorages, vol)
+	}
+	return volumeStorages, nil
+}
+
+func (k *kataAgent) createSMBVolumeObject(c *Container, m Mount) (*grpc.Storage, error) {
+	vol := &grpc.Storage{}
+	vol.Source = m.Source
+	vol.MountPoint = m.Destination
+	vol.Fstype = m.Type
+	vol.Options = m.Options
+	// Todo: replace this with Confidential Data Hub driver
+	vol.Driver = kata9pDevType // 9p as of now calls a default storage handler in kata agent that does a baremount with these options
```

Comment:
Is my understanding below correct?

We specify 9p here for now because the 9p agent handler simply calls mount(2) on this `vol` and that gets us the mount we want, without agent changes. Regardless we'll still need the kernel module in the guest in this PR or a subsequent one (can we do it in this PR?). In the future we'll integrate with the Confidential Data Hub (incl. agent changes) to upstream.

Reply:
Yes, the understanding is correct wrt 9p storage handler. The 9p handler does the mount and for now removes introducing a new storage handler for CIFS in agent. That said, we need to take care that the behavior of 9p handler does not change causing the CIFS functionality to break. In future, we will have to introduce a new handler integrated with CDH, yes.

Reply:
Why not make the Agent change in this PR?

Reply:
I have added a SMBHandler to agent.